src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc

```diff
@@ -165,168 +191,233 @@ const std::string LightName(const smart_objects::SmartObject& light_name) {
-                        << " is missed in RemoteControl capabilities");
-      return false;
-    }
-    if (!capabilities[caps_key].asBool()) {
-      LOG4CXX_DEBUG(logger_,
-                    "Capability "
-                        << caps_key
-                        << " is switched off in RemoteControl capabilities");
-      return false;
+
+    const capabilitiesStatus status_item_capability =
+        GetItemCapability(capabilities,
+                          mapping,
+                          request_parameter,
+                          mobile_apis::Result::READ_ONLY);
+
+    if (capabilitiesStatus::success != status_item_capability) {
+      return std::make_pair(message_params::kLightState,
+                            status_item_capability);
     }
   }
-  return true;
+
+  return std::make_pair("", capabilitiesStatus::success);
 }
 
-bool CheckLightNameByCapabilities(
+/**
+ * @brief Check whether the light name exists in capabilities
+ * @param smart object of capabilities_status
+ * @param smart object of light_data
+ * @return pair of state and capability status - ModuleCapability
+ */
+ModuleCapability GetLightNameCapabilities(
     const smart_objects::SmartObject& capabilities_status,
     const smart_objects::SmartObject& light_data) {
   LOG4CXX_AUTO_TRACE(logger_);
   auto it = capabilities_status.asArray()->begin();
+
   for (; it != capabilities_status.asArray()->end(); ++it) {
     const smart_objects::SmartObject& so = *it;
     const int64_t current_id = so[message_params::kName].asInt();
+
     if (current_id == light_data[message_params::kId].asInt()) {
-      return CheckLightDataByCapabilities(so, light_data);
+      return GetLightDataCapabilities(so, light_data);
     }
   }
-  LOG4CXX_DEBUG(logger_, "There is no such light name in capabilities");
-  return false;
-}
 
-bool CheckRadioBandByCapabilities(
-    const smart_objects::SmartObject& capabilities_status,
-    const smart_objects::SmartObject& request_parameter) {
-  mobile_apis::RadioBand::eType radio_band =
-      static_cast<mobile_apis::RadioBand::eType>(request_parameter.asUInt());
-  if (mobile_apis::RadioBand::XM == radio_band) {
-    if (!capabilities_status.keyExists(strings::kSiriusxmRadioAvailable)) {
-      LOG4CXX_DEBUG(logger_,
-                    "Capability "
-                        << strings::kSiriusxmRadioAvailable
-                        << " is missed in RemoteControl capabilities");
-      return false;
-    }
-    if (!capabilities_status[strings::kSiriusxmRadioAvailable].asBool()) {
-      LOG4CXX_DEBUG(logger_,
-                    "Capability "
-                        << strings::kSiriusxmRadioAvailable
-                        << " is switched off in RemoteControl capabilities");
-      return false;
-    }
-  }
-  return true;
+  LOG4CXX_DEBUG(logger_, "There is no such light name in capabilities");
+  return std::make_pair(message_params::kLightState,
+                        capabilitiesStatus::missedLightName);
+  ;
```

Comment:
Remove extra semicolon

Reply:
@JackLivio  fixed ( amended to previous commit )